File: P02 - Kelas Objek/Java/Bag. kedua/DVD.java
package Praktikum.P02;

import java.util.Scanner;

public class DVD {
    public String judul;
    public String jenis;
    public int harga;
    public int jumlahStock;
    public float nilaiStock;

    public DVD(String judul, String jenis, int harga, int jumlahStock) {
        this.judul = judul;
        this.jenis = (jenis.equalsIgnoreCase("film") || jenis.equalsIgnoreCase("musik") || jenis.equalsIgnoreCase("game")) ? jenis : "film";
        this.harga = harga;
        this.jumlahStock = jumlahStock;
        this.nilaiStock = harga * jumlahStock;
    }

    public void tampilkan() {
        System.out.println("Judul        : " + judul);
        System.out.println("Jenis        : " + jenis);
        System.out.println("Harga        : " + harga);
        System.out.println("Jumlah Stock : " + jumlahStock);
        System.out.println(toString());
    }

    public int tambahStock(int jumlah) {
        if (jumlah > 0) {
            jumlahStock += jumlah;
        }
        return jumlahStock;
    }

    public float penjualan(int jmlJual) {
        if (jmlJual > 0 && jmlJual <= jumlahStock) {
            jumlahStock -= jmlJual;
            return harga * jmlJual;
        } else {
            return 0;
        }
    }

    public float nilaiStock() {
        nilaiStock = harga * jumlahStock;
        return nilaiStock;
    }

    public void tampilkanNilaiStock() {
        System.out.println("Nilai stock dari DVD ini adalah sebesar Rp" + nilaiStock + ",-");
    }

    public String toString() {
        return "Nilai stock dari DVD ini adalah sebesar Rp" + nilaiStock + ",-";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        DVD d1 = new DVD("Paddle Pop Begins", "film", 10000, 12);

        d1.tampilkan();

        int pilih = -1; 
        while (pilih != 0) {
            System.out.println("Apa yang ingin dilakukan?");
            System.out.println("1. Menambah stock DVD");
            System.out.println("2. Menjual DVD");
            System.out.println("3. Tampilkan nilai stock");
            System.out.println("0. Keluar\n");

            pilih = scanner.nextInt();
            if (pilih == 1) {
                System.out.print("Berapa stock yang ditambah? ");
                int ditambah = scanner.nextInt();
                d1.tambahStock(ditambah);
                System.out.println("Stock DVD saat ini adalah " + d1.jumlahStock + " buah");
            } else if (pilih == 2) {
                System.out.print("Berapa stock yang terjual? ");
                int terjual = scanner.nextInt();
                float pendapatan = d1.penjualan(terjual);
                if (pendapatan > 0) {
                    System.out.println("Terjual DVD " + terjual + " buah dengan harga Rp" + pendapatan + ",-");
                } else {
                    System.out.println("Stok tidak mencukupi untuk penjualan.");
                }
            } else if (pilih == 3) {
                d1.tampilkanNilaiStock();
            } else if (pilih == 0) {
                System.out.println("Terima kasih. Sampai jumpa!");
            } else {
                System.out.println("Pilihan tidak valid.");
            }
        }

        scanner.close();
    }
}

File: P02 - Kelas Objek/Python/Bag. kedua/DVD.py
# File: DVD.py
# Penulis : Elmosius Suli
# Membuat kelas DVD

class DVD:
    def __init__(self, judul:str, jenis:str, harga:int, jumlahStock:int) -> None:
        self.judul = judul
        self.jenis = jenis if jenis in {"film", "musik", "game"} else "film"
        self.harga = harga
        self.jumlahStock = jumlahStock
        return

    def tampilkan(self) -> None:
        print("="*40)
        print("Judul        :", self.judul)
        print("Jenis        :", self.jenis)
        print("Harga        :", self.harga)
        print("Jumlah Stock :", self.jumlahStock)
        print(self.toString())

    def tambahStock(self, jumlah:int) -> int:
        self.jumlahStock += jumlah
        return self.jumlahStock

    def penjualan(self, jmlJual:int) -> float:
        self.jumlahStock -= jmlJual
        return self.harga * jmlJual

    def nilaiStock(self) -> float:
        self.nilaiStock = self.harga * self.jumlahStock
        return self.nilaiStock

    def toString(self) -> str:
        return f"Nilai stock dari DVD ini adalah sebesar Rp{self.nilaiStock()},-" 

## Program utama ##
def main():
    d1 = DVD("Paddle Pop Begins", "movie", 10000, 12)
    d1.tampilkan()
    d2 = DVD("The Sims", "game", 15000, 5)
    d2.tampilkan()
    print("="*40)

    status = int(input("Kamu adalah (1. Penjual, 2. Pembeli) : "))

    if (status == 1):
        print("Apa yang ingin dilakukan?")
        print("1. Menambah stock DVD")
        print("2. Menjual DVD")
        print("0. Keluar")
        pilih = int(input("Silakan masukkan angka : "))

        while (pilih != 0):
            if (pilih == 1):
                ditambah = int(input("Berapa stock yang ditambah? "))
                print(f"Stock DVD saat ini adalah {d1.tambahStock(ditambah)} buah")
            elif (pilih == 2):
                terjual = int(input("Berapa stock yang terjual? "))
                print(f"Terjual DVD {terjual} buah dengan harga Rp{d1.penjualan(terjual)},-")

            print("--------------------------")
            print("Apa yang ingin dilakukan?")
            print("1. Menambah stock DVD")
            print("2. Menjual DVD")
            print("0. Keluar")
            pilih = int(input("Silakan masukkan angka : "))
    
    elif (status == 2):
        harga = 0
        jml1 = 0
        jml2 = 0

        print(f"1. {d1.judul}")
        print(f"2. {d2.judul}")
        print("0. Keluar")
        pilihDVD = int(input("Masukkan kode DVD yang ingin dibeli: "))
        while (pilihDVD != 0):
            jumlah = int(input("Berapa banyak yang akan dibeli? "))
            if (pilihDVD == 1):
                harga += d1.harga * jumlah
                jml1 += jumlah
            elif (pilihDVD == 2):
                harga += d2.harga * jumlah
                jml2 += jumlah

            print(f"1. {d1.judul}")
            print(f"2. {d2.judul}")
            print("0. Keluar")
            pilihDVD = int(input("Masukkan kode DVD yang ingin dibeli: "))      

        print("Daftar DVD yang dibeli :")
        if (jml1 != 0):
            print(f"{d1.judul} sebanyak {jml1} buah")
        if (jml2 != 0):
            print(f"{d2.judul} sebanyak {jml2} buah")
        print(f"Total harga yang harus dibayar = Rp{harga},-")

if __name__ == '__main__':
    main()